internal/internal_workflow_client_test.go

```diff
@@ -1443,3 +1703,150 @@ func (m *PartialCancelRequestMatcher) Matches(a interface{}) bool {
+				ID:                              workflowID,
+				TaskList:                        tasklist,
+				ExecutionStartToCloseTimeout:    0 * time.Second, // this causes error
+				DecisionTaskStartToCloseTimeout: 5 * time.Second,
+				DelayStart:                      0 * time.Second,
+				JitterStart:                     0 * time.Second,
+			},
+			workflowFunc: func(ctx Context) {},
+			wantErr:      "missing or invalid ExecutionStartToCloseTimeout",
+		},
+		{
+			name: "negative DecisionTaskStartToCloseTimeout",
+			options: StartWorkflowOptions{
+				ID:                              workflowID,
+				TaskList:                        tasklist,
+				ExecutionStartToCloseTimeout:    10 * time.Second,
+				DecisionTaskStartToCloseTimeout: -1 * time.Second, // this causes error
+				DelayStart:                      0 * time.Second,
+				JitterStart:                     0 * time.Second,
+			},
+			workflowFunc: func(ctx Context) {},
+			wantErr:      "negative DecisionTaskStartToCloseTimeout provided",
+		},
+		{
+			name: "negative DelayStart",
+			options: StartWorkflowOptions{
+				ID:                              workflowID,
+				TaskList:                        tasklist,
+				ExecutionStartToCloseTimeout:    10 * time.Second,
+				DecisionTaskStartToCloseTimeout: 5 * time.Second,
+				DelayStart:                      -1 * time.Second, // this causes error
+				JitterStart:                     0 * time.Second,
+			},
+			workflowFunc: func(ctx Context) {},
+			wantErr:      "Invalid DelayStart option",
+		},
+		{
+			name: "negative JitterStart",
+			options: StartWorkflowOptions{
+				ID:                              workflowID,
+				TaskList:                        tasklist,
+				ExecutionStartToCloseTimeout:    10 * time.Second,
+				DecisionTaskStartToCloseTimeout: 5 * time.Second,
+				DelayStart:                      0 * time.Second,
+				JitterStart:                     -1 * time.Second, // this causes error
+			},
+			workflowFunc: func(ctx Context) {},
+			wantErr:      "Invalid JitterStart option",
+		},
+		{
+			name: "invalid workflow func",
+			options: StartWorkflowOptions{
+				ID:                              workflowID,
+				TaskList:                        tasklist,
+				ExecutionStartToCloseTimeout:    10 * time.Second,
+				DecisionTaskStartToCloseTimeout: 5 * time.Second,
+				DelayStart:                      0 * time.Second,
+				JitterStart:                     0 * time.Second,
+			},
+			workflowFunc: func(ctx Context, a, b int) {}, // this causes error because args not provided
+			wantErr:      "expected 2 args for function",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			mockCtrl := gomock.NewController(t)
+			service := workflowservicetest.NewMockClient(mockCtrl)
+			wc, ok := NewClient(service, domain, &ClientOptions{
+				Identity: "test-identity",
+			}).(*workflowClient)
+
+			if !ok {
+				t.Fatalf("expected NewClient to return a *workflowClient, but got %T", wc)
+			}
```

Comment:
```suggestion
			wc := NewClient(service, domain, &ClientOptions{
				Identity: "test-identity",
			}).(*workflowClient)
```
the panic already tells you the same info, and even slightly more.  simpler to just let it bubble.

Reply:
I am guessing panic would not give me the actual type which I'm explicitly adding in log message with `%T`. Either way it's a minor thing, leaving as is.

Reply:
It does. You get the interface type, the concrete type in the interface, and the target type-asserted type.